packages/admin-next/dashboard/src/hooks/api/campaigns.tsx

```diff
@@ -105,3 +107,19 @@ export const useDeleteCampaign = (
     ...options,
   })
 }
+
+export const useAddOrRemoveCampaignPromotions = (
+  id: string,
+  options?: UseMutationOptions<AdminCampaignResponse, Error, LinkMethodRequest>
+) => {
+  return useMutation({
+    mutationFn: (payload) =>
+      client.campaigns.addOrRemovePromotions(id, payload),
+    onSuccess: (data, variables, context) => {
+      queryClient.invalidateQueries({ queryKey: campaignsQueryKeys.detail(id) })
```

Comment:
suggestion: this will invalidate only route detail rout with id and different query params wont be taken into account (you can use `details()`)